File: 3-arrays/2-array-types/main.java
// Array Types - Java Example

public class ArrayTypes {

    public static void main(String[] args) {
        // 1D array
        int[] arr1d = {1, 2, 3};

        // 2D array
        int[][] arr2d = {{1, 2}, {3, 4}};

        System.out.print("1D array: ");
        for (int num : arr1d) {
            System.out.print(num + " ");
        }
        System.out.println();

        System.out.println("2D array:");
        for (int[] row : arr2d) {
            for (int num : row) {
                System.out.print(num + " ");
            }
            System.out.println();
        }
    }
}

File: 3-arrays/2-array-types/main.cpp
// Array Types - C++ Example

#include <iostream>

int main() {
    int arr_1d[] = {1, 2, 3};
    int arr_2d[2][2] = {{1, 2}, {3, 4}};

    std::cout << "1D array: ";
    for (int i = 0; i < 3; i++) {
        std::cout << arr_1d[i] << " ";
    }
    std::cout << std::endl;

    std::cout << "2D array:" << std::endl;
    for (int i = 0; i < 2; i++) {
        for (int j = 0; j < 2; j++) {
            std::cout << arr_2d[i][j] << " ";
        }
        std::cout << std::endl;
    }
    return 0;
}
